core/src/main/java/jenkins/mvn/FilePathGlobalSettingsProvider.java

```diff
@@ -32,34 +37,45 @@ public String getPath() {
     }
 
     @Override
-    public FilePath supplySettings(AbstractBuild<?, ?> build, TaskListener listener) {
+    public FilePath supplySettings(Run<?, ?> run, FilePath workspace, TaskListener listener) throws IOException, InterruptedException {
         if (StringUtils.isEmpty(path)) {
             return null;
         }
 
         try {
-            EnvVars env = build.getEnvironment(listener);
-            String targetPath = Util.replaceMacro(this.path, build.getBuildVariableResolver());
-            targetPath = env.expand(targetPath);
+            EnvVars env = run.getEnvironment(listener);
 
-            if (IOUtils.isAbsolute(targetPath)) {
-                return new FilePath(new File(targetPath));
-            } else {
-                FilePath mrSettings = build.getModuleRoot().child(targetPath);
-                FilePath wsSettings = build.getWorkspace().child(targetPath);
-                try {
+            FilePath result;
+            if (run instanceof AbstractBuild) {
+                AbstractBuild build = (AbstractBuild) run;
+                String targetPath = Util.replaceMacro(this.path, build.getBuildVariableResolver());
+                targetPath = env.expand(targetPath);
+                if (IOUtils.isAbsolute(targetPath)) {
+                    return new FilePath(new File(targetPath));
+                } else {
+                    FilePath mrSettings = build.getModuleRoot().child(targetPath);
+                    FilePath wsSettings = build.getWorkspace().child(targetPath);
                     if (!wsSettings.exists() && mrSettings.exists()) {
-                        wsSettings = mrSettings;
+                        result = mrSettings;
+                    } else {
+                        result = wsSettings;
                     }
-                } catch (Exception e) {
-                    throw new IllegalStateException("failed to find settings.xml at: " + wsSettings.getRemote());
                 }
-                return wsSettings;
+            } else {
+                String targetPath = env.expand(path);
+                if (IOUtils.isAbsolute(targetPath)) {
+                    result = new FilePath(new File(targetPath));
+                } else {
+                    result = workspace.child(targetPath);
+                }
             }
+            LOGGER.log(Level.FINE, "Supply Maven global settings.xml file {0} to {1}", new Object[]{result, run});
+            return result;
         } catch (Exception e) {
-            throw new IllegalStateException("failed to prepare global settings.xml");
+            throw new IllegalStateException(
```

Comment:
Delete—I thought the whole reason to allow the method to throw `IOException | InterruptedException` was so that we would need to wrap in a `RuntimeException`.